src/lib.rs

```diff
@@ -243,6 +246,25 @@ fn get_num_cpus() -> usize {
     cpus as usize
 }
 
+
+#[cfg(target_os = "macos")]
+fn get_num_physical_cpus() -> usize {
+    let mut cpus: i32 = 0;
+    let mut cpus_size = std::mem::size_of_val(&cpus);
+
+    let sysctl_name = CString::new("hw.physicalcpu").unwrap();
```

Comment:
Suggestion: instead of including `CString`, you could use a [bytes string literal](https://doc.rust-lang.org/reference/tokens.html#byte-string-literals), i.e. `b"hw.physicalcpu\0"`